optuna/distributions.py

```diff
@@ -217,7 +217,11 @@ def __init__(self, low, high, q):
     def single(self):
         # type: () -> bool
 
-        return self.low == self.high
+        if self.low == self.high:
+            return True
+        if (self.high - self.low) < self.q:
+            return True
+        return False
```

Comment:
Now I'm considering to close this pull request if we should support the following case.

```python
>>> single(0.2, 0.1, 0.1)
False
>>> single(0.3, 0.2, 0.1)
True
>>> 0.3 - 0.2
0.09999999999999998
```

Reply:
In the discussion of core-devs, we decided to use decimal module.